src/Socket_p.h

```diff
@@ -352,7 +352,7 @@ namespace Arcus
 
         {
             std::lock_guard<std::mutex> lk(receiveQueueMutexBlock);
```

Comment:
Since the (unsynchronized) boolean has now become an actual `condition_variable`, I think these locks have become unnescesary; because we just wait-until-notify on it, instead of setting it. (Note that `lock-guard` only locks the mutex within the scope of its creation.) Spurious locking may degrade performance (only a little in this case, but still).